roles/chameleon_mariadb/tasks/main.yml

```diff
@@ -19,6 +19,37 @@
     dest: "{{ mariadb_backup_cron_script }}"
   when: inventory_hostname_short == "dev"
 
+- name: Create directory to mount to aws-cli docker container for copying config and backup files to container
+  file:
+    path: /etc/awsconf
+    state: directory
+  when: inventory_hostname_short == "admin01" or
```

Comment:
A role should not need to understand which hosts it is running on. The inventory decides how to map roles to hosts or groups of hosts. You can imagine that this approach becomes difficult to scale to TACC=>UC, many of the templates would need duplicating and tasks too. However, almost all of what you have is very close to being entirely generic!

What I recommend:
- Create a new variable to hold the value of the backup destination (the Ceph S3 endpoint). Use this variable when templating the backup script.
- Consider setting aws credentials as environment variables to the container (if possible), this will remove the need for that additional file just to hold the creds
- Rename `tacc_ceph_backup_access_key` to just `mariadb_ceph_backup_access_key`. Typically variables are prefixed w/ the name of the role they are tied to. This also allows each site to set a different value for this in site-configs.

Reply:
Thanks Jason, I've made all the updates.